### PR TITLE
Update for elm 0.18

### PR DESCRIPTION
**Pull request by dillonkearns**

### CODE REVIEW

File: src/GildedRose.elm [hunk 1]
```diff
@@ -16,7 +16,7 @@ updateQuality =
 
 updateQualityItem (Item name sellIn quality) =
     let
-        quality' =
+        quality_ =
```
Comment (pierrebeitz) [inline src/GildedRose.elm:19]: as the reason for deprecating `'`s was to force better names i guess we should aim for something like `updatedQuality`. What do you think?

File: elm-package.json [hunk 2]
```diff
@@ -8,9 +8,9 @@
     ],
     "exposed-modules": [],
     "dependencies": {
-        "elm-community/elm-test": "2.1.0 <= v < 3.0.0",
-        "elm-lang/core": "4.0.5 <= v < 5.0.0",
-        "elm-lang/html": "1.1.0 <= v < 2.0.0"
+        "elm-community/elm-test": "3.1.0 <= v < 4.0.0",
+        "elm-lang/core": "5.1.1 <= v < 6.0.0",
+        "elm-lang/html": "2.0.0 <= v < 3.0.0"
```
Comment (pierrebeitz) [inline elm-package.json]: Do we actually need `elm-lang/html`. I think i put it in here by mistake. Could you please verify?

**Comment by pierrebeitz** [comment]
thank you so much, @dillonkearns!

**Review by pierrebeitz — COMMENTED** [review]
i'm fine with introducing yarn. but please update the readme accordingly!

File: src/GildedRose.elm (hunk 1 above)
Comment (dillonkearns) [reply src/GildedRose.elm:19]: That's definitely a good practice, although this is a legacy code refactoring exercise so having unclear names could be good? I don't have a strong opinion, could go either way.

File: elm-package.json (hunk 2 above)
Comment (dillonkearns) [reply elm-package.json]: That's true, and elm-test isn't needed here either.

**Comment by dillonkearns** [comment]
@pierrebeitz sorry for the delay, finally addressed your comments. As for yarn test, I removed the lock file for now to avoid confusion, so people can choose to use it or not and it shouldn't be an issue.

File: src/GildedRose.elm (hunk 1 above)
Comment (pierrebeitz) [reply src/GildedRose.elm:19]: oh! good point. so be it!

**Comment by pierrebeitz** [comment]
looks good to me! thanks a lot!

**Comment by dillonkearns** [comment]
👍 neat, and thanks for sharing the exercise, @pierrebeitz!

**Pull request merged**